Add DublinCore metadata to Book page

jakkarth opened the pull request:
Put some dublin core metadata in the book pages for bibliographic tools like Zotero.

<!--
Thanks for contributing! This template has some checkboxes that help keep track of what changes go into a release.

To check (tick) a list item, replace the space between square brackets with an x, like this:

- [x] I have checked the box

You can find more information and tips for BookWyrm contributors at https://docs.joinbookwyrm.com/contributing.html
-->
## Description
<!--
Describe what your pull request does here
-->
Add a template snippet to output DublinCore-compatible metadata for use by bibliometric tools like Zotero, and reference it from the Book template.

<!--
For pull requests that relate or close an issue, please include them
below.  We like to follow [Github's guidance on linking issues to pull requests](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

For example having the text: "closes #1234" would connect the current pull
request to issue 1234.  And when we merge the pull request, Github will
automatically close the issue.
-->

Closes #3107 

## What type of Pull Request is this?
<!-- Check all that apply -->

- [ ] Bug Fix
- [x] Enhancement
- [ ] Plumbing / Internals / Dependencies
- [ ] Refactor

## Does this PR change settings or dependencies, or break something?
<!-- Check all that apply -->

- [ ] This PR changes or adds default settings, configuration, or .env values
- [ ] This PR changes or adds dependencies
- [ ] This PR introduces other breaking changes

### Details of breaking or configuration changes (if any of above checked)


## Documentation
<!--
Documentation for users, admins, and developers is an important way to keep the BookWyrm community welcoming and make Bookwyrm easy to use.
Our documentation is maintained in a separate repository at https://github.com/bookwyrm-social/documentation
-->
Not related to this PR, but as a first-time contributor to this project I've certainly had some struggles getting started. The documentation at https://docs.joinbookwyrm.com/install-dev.html don't indicate which branch in the main repo to branch from, several required values in .env aren't mentioned, and there's some kind of caching issue where my changes only showed up about 30% of the time unless I tore the whole environment down and restarted it. Clarifying some of these points in the developer documentation would make contributions easier.

If the above documentation does already exist somewhere, I apologize for not being able to find it. Adding a link to it from the page I mentioned above might be helpful!
<!-- Check all that apply -->

- [ ] New or amended documentation will be required if this PR is merged
- [ ] I have created a matching pull request in the Documentation repository
- [ ] I intend to create a matching pull request in the Documentation repository after this PR is merged

<!-- Amazing! Thanks for filling that out. Your PR will need to have passing tests and happy linters before we can merge
You will need to check your code with `black`, `pylint`, and `mypy`, or `./bw-dev formatters`
-->

### Tests
<!-- Check one -->

- [ ] My changes do not need new tests
- [ ] All tests I have added are passing
- [ ] I have written tests but need help to make them pass
- [x] I have not written tests and need help to write them

I'm assuming I need tests but I'm not familiar enough with this ecosystem to know where to start.

### Other

This is my first time contributing to this project. I've been wanting an easy way to get metadata into Zotero for my books, and since there's an open issue for it tagged as "good first issue" I thought I'd take a crack at it. I am very open to feedback!

hughrun reviewed (changes requested):
This is great!

Thankyou for your thoughtful comments on how we can improve documentation for first-time contributors. Our docs are a work in progress. I'll take a look at the issues you've mentioned and may come back to you to check what specific fixes would be helpful or whether my amendments fix the problems you have highlighted.

This looks pretty good to me - just a couple of issues I noticed, and a suggestion for making it even better.

jakkarth commented:
Thank you for the encouragement and feedback! I've implemented all three suggestions and updated the PR, rebasing it on the current main branch. Please let me know if there's anything else needed :)
